Sources/xcodeproj/Objects/Project/PBXObject.swift

```diff
@@ -13,6 +13,9 @@ public class PBXObject: Hashable, Decodable, Equatable, AutoEquatable {
     /// The object reference in the project that contains it.
     let reference: PBXObjectReference
 
+    /// Used to differentiate this object from other equatable ones for the purpose of reference generation
+    public var identifier: String?
```

Comment:
I think it should be enough with the logic that was added to the `ReferenceGenerator`. Could you point me to a concrete example on XcodeGen to better understand why this might be necessary?

Reply:
It’s required in the linked XcodeGen PR https://github.com/yonaskolb/XcodeGen/pull/422/files#diff-2f3c621620623f55e7ff42cc32198a10R136
Without this then some file references would be duplicates and get the numbers appended, which also isn’t deterministic as they can’t be sorted in the group they are in, so would always cause diffs

Reply:
This context is something we lost the ability to add with the move to automatic reference generation. With saying that I think if this stays around it should be renamed to `context`

Reply:
> Without this then some file references would be duplicates and get the numbers appended, which also isn’t deterministic as they can’t be sorted in the group they are in, so would always cause diffs

I see. I was wondering if we could somehow make that scenario, two targets with the same name, deterministic. For example, those targets have a fixed position in the project they belong to. What about using that?

I think we should aim for exposing as less as possible to make the API easy to use. If we add properties like this one, `identifier` or `context`, there's one more thing the consumer of the API needs to reason about.

Reply:
By position do you mean sort? Because they are equatable you can’t sort them. Or did you have another idea?

I totally agree about not exposing too much public api. This is a totally optional property though. I can add more documentation explaining that

Reply:
> By position do you mean sort? Because they are equatable you can’t sort them. Or did you have another idea?

I mean that their position [in the project](https://github.com/tuist/xcodeproj/blob/master/Sources/xcodeproj/Objects/Project/PBXProject.swift#L93) can be used as a context. Do you think that'd work?

> I totally agree about not exposing too much public api. This is a totally optional property though. I can add more documentation explaining that

Worst case we need to have that property we should definitively docucument it 👍